### DISCUSSION TITLE
What if I want to get a non-empty array?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#2850] [Questions] [answered]
I only want to retrieve the ones that have data.
Using supbase-js.
How can I use Filter?

![スクリーンショット 2021-08-16 19 24 12](https://user-images.githubusercontent.com/42270511/129549529-3c6de4bc-95e8-4ff9-b002-ee52cf9d0d27.png)

**Comment by steve-chavez** [comment, marked as answer]
In SQL, to get all the empty arrays you can do:

```sql
select * from tbl where images <@ '{}';
```

Negating that filter would give you all the non-empty arrays.

`<@` is exposed in PostgREST as the [cd(contained)](https://postgrest.org/en/v8.0/api.html#operators) operator, which in turn can be used and negated in `supabase-js` with [not](https://supabase.io/docs/reference/javascript/not):

```js
const { data, error } = await supabase
  .from('tbl')
  .select('*')
  .not('images', 'cd', '{}')
```

**Reply by fergusmeiklejohn** [reply]
We should put this one in the docs, it can also help explain how to build similar queries

**Reply by ru88s** [reply]
It's perfect.
Thank you very much!

**Reply by steve-chavez** [reply]
@fergusmeiklejohn Good idea. The doc would show how to go from SQL to the API layer for a query.
